### PR TITLE
DFPL-2040 (base:2023)Temporary confidential document tab

### PR DESCRIPTION
**Pull request by so99y**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-2040

### Change description ###
Add a temporary confidential document tab to show the confidential documents.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json [hunk 1]
```diff
@@ -4157,6 +4157,13 @@
     "UserRole": "caseworker-publiclaw-judiciary",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
```
Comment (jamesrferguson1) [inline ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json:4161]: do the judges not also need permissions for confidentialDocumentListPageHMCTS?

File: ccd-definition/CaseTypeTab/CareSupervision/0_ApplicationTasks.json [hunk 2]
```diff
@@ -14,8 +14,6 @@
     "CaseTypeID": "CARE_SUPERVISION_EPO",
     "Channel": "CaseWorker",
     "TabID": "Tasks",
-    "TabLabel": "Start application",
```
Comment (jamesrferguson1) [inline ccd-definition/CaseTypeTab/CareSupervision/0_ApplicationTasks.json:17]: why are we removing this?

File: ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json (hunk 1 above)
Comment (so99y) [reply ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json:4161]: Good catch.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/aggregator/BundleViewAggregator.java [hunk 3]
```diff
@@ -53,4 +94,40 @@ public List<DocumentContainerView> getDocumentBundleViews(
...
+            .fileName(filename)
+            .confidentialLevel(level == CTSC ? "Restrict to HMCTS staff"
+                : "Restrict to the LA, Cafcass and HMCTS staff")
+            .uploaderType(md.getUploaderType() == null ? "unknown" : md.getUploaderType().name())
+            .documentType(StringUtils.replace(documentType.getDescription(), "└─ ",  ""))
+            .build();
+    }
+
+    public List<DocumentView> getConfidentialDocumentBundleViews(CaseData caseData, DocumentViewType view) {
+        DocumentType[] documentTypes = new DocumentType[] {CASE_SUMMARY, POSITION_STATEMENTS, POSITION_STATEMENTS_CHILD,
```
Comment (jamesrferguson1) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/aggregator/BundleViewAggregator.java:112]: would DocumentType.values() give us this?

Comment (so99y) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/aggregator/BundleViewAggregator.java:112]: The confidential document tab is a temporary solution. According to Sara's email, if the confidential folder was introduced, those document types would be removed from the view.

And not all the document types are fitted to the generic logic (e.g. Placement Response).

**Review by jamesrferguson1 — DISMISSED** [review]

**Pull request closed without merging**